dialect.go

```diff
@@ -1,364 +1,41 @@
 package goose
 
 import (
-	"database/sql"
 	"fmt"
+
+	"github.com/pressly/goose/v3/internal/dialect"
 )
 
-// SQLDialect abstracts the details of specific SQL dialects
-// for goose's few SQL specific statements
-type SQLDialect interface {
-	createVersionTableSQL() string // sql string to create the db version table
-	insertVersionSQL() string      // sql string to insert the initial version table row
-	deleteVersionSQL() string      // sql string to delete version
-	migrationSQL() string          // sql string to retrieve migrations
-	dbVersionQuery(db *sql.DB) (*sql.Rows, error)
+func init() {
+	store, _ = dialect.NewStore(dialect.Postgres, TableName())
 }
 
-var dialect SQLDialect = &PostgresDialect{}
-
-// GetDialect gets the SQLDialect
-func GetDialect() SQLDialect {
```

Comment:
This is the ONLY breaking change in this whole PR. But it's also inconsequential, because users can do anything with the `SQLDialect` since all the methods are private to this package.